experimental/framework/fluid-static/src/containerCode.ts

```diff
@@ -115,6 +122,25 @@ export class RootDataObject
         super.dispose();
     }
 
+    public setContainer(container: Container) {
+        this._container = container;
+    }
+
+    public close() {
+        assert(this._container !== undefined, "The container has not been set by the client");
+        this._container.close();
+    }
+
+    public get closed() {
+        assert(this._container !== undefined, "The container has not been set by the client");
```

Comment:
To avoid repeating the assert it can be nice to give yourself a private accessor like

```typescript
private get container() {
   assert(this._container !== undefined, "The container has not been set by the client");
   return this._container;
}
```

And then you can just use `this.container` throughout.